### PR TITLE
Upgrade connection matrix heuristic

### PR DESCRIPTION
**Pull request by AdamWiktor**

### CODE REVIEW

File: src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/ConnectionMatrixShortestPathTreeRWH.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        spt();
+    }
+
+    @Override
+    public double estimateRemainingWeight(State s) {
+        Point source = data.mapToPoint(s.getVertex());
+        if (!data.getBoundaries().contains(source) || !data.getBoundaries().contains(destination)) {
+            return fallback.estimateRemainingWeight(s);
+        }
+        return 0.;
```
Comment (Kamyki) [inline src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/ConnectionMatrixShortestPathTreeRWH.java]: !!!

File: src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/ConnectionMatrixShortestPathTreeRWH.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        }
+        visited[current.getPoint().getI()][current.getPoint().getJ()] = true;
+        estimation[current.getPoint().getI()][current.getPoint().getJ()] = current.getPoint().equals(destination) ? 0.f : current.getWeight();
+        addNeighbors(current);
+    }
+
+    private void addNeighbors(PointWithWeight current) {
+        data.getNeighbors(current.getPoint())
+                .filter(neighbor -> !visited[neighbor.getPoint().getI()][neighbor.getPoint().getJ()])
+                .map(neighbor -> new PointWithWeight(neighbor.getPoint(), current.getWeight() + data.getCost(current.getPoint(), neighbor.getDirection())))
```
Comment (Kamyki) [inline src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/ConnectionMatrixShortestPathTreeRWH.java]: data.getCost(neighbour.getPoint(), neighbor.getDirection().opposite()) ?
http://mathcenter.oxford.emory.edu/site/cs171/dijkstrasShortestPathAlgorithm/ says reversing edges is correct

**Review by Kamyki — APPROVED** [review]

File: src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/ConnectionMatrixShortestPathTreeRWH.java (hunk 1 above)
Comment (AdamWiktor) [reply src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/ConnectionMatrixShortestPathTreeRWH.java]: done

File: src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/ConnectionMatrixShortestPathTreeRWH.java (hunk 2 above)
Comment (AdamWiktor) [reply src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/ConnectionMatrixShortestPathTreeRWH.java]: done

**Pull request merged**